### PR TITLE
example(core): Add example for TypeScript transpiling via deno_ast

### PR DESCRIPTION
**Pull request by andreubotella**

### CODE REVIEW

File: core/examples/ts_module_loader.rs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+  let mut js_runtime = JsRuntime::new(RuntimeOptions {
+    module_loader: Some(Rc::new(TypescriptModuleLoader)),
+    ..Default::default()
+  });
+
+  let main_module = {
+    let mut path = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+    path.push("examples/ts_module_loader.ts");
+    resolve_path(path.to_str().unwrap())?
+  };
```
Comment (bartlomieju) [inline core/examples/ts_module_loader.rs:105]: Could you align this example to `core/examples/fs_module_loader.rs` to allow executing arbitrary TS file specified as a CLI arg?

Comment (andreubotella) [reply core/examples/ts_module_loader.rs]: Done.

File: core/examples/ts_module_loader.rs [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    _is_dyn_import: bool,
+  ) -> Pin<Box<ModuleSourceFuture>> {
+    let module_specifier = module_specifier.clone();
+    async move {
+      let path = module_specifier
+        .to_file_path()
+        .map_err(|_| anyhow!("Only file: URLs are supported."))?;
+
+      let media_type = MediaType::from(&path);
+      let (module_type, do_transpile) = match MediaType::from(&path) {
```
Comment (bartlomieju) [inline core/examples/ts_module_loader.rs:52]:
```suggestion
      let (module_type, should_transpile) = match MediaType::from(&path) {
```

File: core/examples/ts_module_loader.rs [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        | MediaType::Dts
+        | MediaType::Dmts
+        | MediaType::Dcts
+        | MediaType::Tsx => (ModuleType::JavaScript, true),
+        MediaType::Json => (ModuleType::Json, false),
+        _ => bail!("Unknown extension {:?}", path.extension()),
+      };
+
+      let code = std::fs::read_to_string(&path)?;
+      let code = if do_transpile {
```
Comment (bartlomieju) [inline core/examples/ts_module_loader.rs:69]:
```suggestion
      let code = if should_transpile {
```

File: core/examples/ts_module_loader.rs [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        }
+        MediaType::Jsx => (ModuleType::JavaScript, true),
+        MediaType::TypeScript
+        | MediaType::Mts
+        | MediaType::Cts
+        | MediaType::Dts
+        | MediaType::Dmts
+        | MediaType::Dcts
+        | MediaType::Tsx => (ModuleType::JavaScript, true),
+        MediaType::Json => (ModuleType::Json, false),
```
Comment (bartlomieju) [inline core/examples/ts_module_loader.rs:64]: Nice 👍

**Review by bartlomieju — COMMENTED** [review]
One more nitpick. Awesome example @andreubotella!

**Review by bartlomieju — APPROVED** [review]
LGTM, very nice, thank you @andreubotella

**Pull request merged**